src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -23,8 +25,10 @@
 )
 
 
-def _aggregation_on_session_status_func_factory(aggregate):
-    def _snql_on_session_status_factory(org_id, session_status, metric_ids, alias=None):
+def _aggregation_on_session_status_func_factory(aggregate) -> Function:
+    def _snql_on_session_status_factory(
+        org_id: int, session_status: str, metric_ids: list[int], alias: Optional[str] = None
```

Comment:
We generally use `List[x]` instead of `list[x]` across our codebase. Is there any specific reason to use `list[x]` Maybe I am missing the semantic difference between the two.

Reply:
> Importing those from typing is deprecated. Due to [PEP 563](https://www.python.org/dev/peps/pep-0563/) and the intention to minimize the runtime impact of typing, this deprecation will not generate DeprecationWarnings. Instead, type checkers may warn about such deprecated usage when the target version of the checked program is signalled to be Python 3.9 or newer. It's recommended to allow for those warnings to be silenced on a project-wide basis.

>    The deprecated functionality will be removed from the typing module in the first Python version released 5 years after the release of Python 3.9.0.